src/ipumspy/ddi.py

```diff
@@ -242,3 +256,64 @@ def get_variable_info(self, name: str) -> VariableDescription:
             ][0]
         except IndexError:
             raise ValueError(f"No description found for {name}.")
+
+    def get_all_types(self, type_format: str, string_pyarrow: bool = False) -> dict:
+        """
+        Retrieve all column types
+
+        Args:
+            type_format: type format. Should be one of ["numpy_type", "pandas_type", "pandas_type_efficient",
+                         "python_type", "vartype"]
+            string_pyarrow: has an effect when True and used with type_format in ["pandas_type", "pandas_type_efficient"].
+             In this case, string types==pd.StringDtype() is replaced with pd.StringDtype(storage='pyarrow').
+
+        Returns:
+            A dict with column names column dtype mapping.
+
+        Examples:
+            Let's see an example of usage with pandas.read_csv:
+
+            >>> from ipumspy import readers
+            >>> reader = readers.read_ipums_ddi('extract_ddi.xml')
+            >>> dataframe_dtypes = reader.get_all_types(type_format='pandas_type', string_pyarrow=False)
+            >>> df = pd.read_csv("extract.csv", dtype=dataframe_dtypes)
```

Comment:
I would suggest updating `_read_microdata()` to accept a data type dict as an optional argument to allow users to use the `readers.read_microdata()` methods rather than reading through pandas directly.